### ISSUE TITLE
Boat Reservation

### ISSUE THREAD

**Issue by ValentaCodes** [ValentaCodes/DiveIn#39]
We need to create the functionality that allows a user (renter ) to reserve and cancel a reservation. This will be accomplished with api route

**Comment by MiahHub** [comment]
Are you thinking that a valid reservation means that for a given boat, boat.renter_id is assigned a renter id and boat.availability = 0 and that a cancelled reservation means boat.renter_id = NULL and boat.availability = 1

**Comment by ValentaCodes** [comment]
Yup that’s what I’m thinking it would look like. Except if boat is canceled
it would just return to true again or in API test 1

**Comment by ValentaCodes** [comment]
Oh wait that what you said disregard what I just sent sorry I read it wrong
at first

**Comment by ValentaCodes** [comment]
You’re definitely on the right track!

**Comment by MiahHub** [comment]
no worries we can discuss in detail.  Thanks!
